packages/studioCMS/src/utils/authhelper.ts

```diff
@@ -15,72 +15,56 @@ type authHelperResponse = {
 };
 
 export default async function authHelper(locals: Locals): Promise<authHelperResponse> {
-	const isLoggedIn = locals.isLoggedIn;
-
-	if (isLoggedIn) {
-		if (locals.dbUser) {
-			const sessionArray = await lucia.getUserSessions(locals.dbUser.id);
-			const id = locals.dbUser.id;
-			const username = locals.dbUser.username;
-			const name = locals.dbUser.name;
-			const email = locals.dbUser.email;
-			const avatar = locals.dbUser.avatar;
-			const githubURL = locals.dbUser.githubURL;
-			const currentUserSession = sessionArray[0];
-			let permissionLevel: 'admin' | 'editor' | 'visitor' | 'unknown' = 'unknown';
-			const permissions = await db
-				.select()
-				.from(Permissions)
-				.where(eq(Permissions.username, username))
-				.get();
-
-			if (permissions) {
-				if (permissions && permissions.username.toLowerCase() === username.toLowerCase()) {
-					if (permissions.rank === 'admin') {
-						permissionLevel = 'admin';
-					}
-					if (permissions.rank === 'editor') {
-						permissionLevel = 'editor';
-					}
-					if (permissions.rank === 'visitor') {
-						permissionLevel = 'visitor';
-					}
-				} else {
-					permissionLevel = 'visitor';
-				}
-			} else {
-				permissionLevel = 'visitor';
-			}
-			return {
-				id,
-				username,
-				name,
-				email,
-				avatar,
-				githubURL,
-				permissionLevel,
-				currentUserSession,
-			};
-		}
-		return {
-			id: '',
-			username: null,
-			name: null,
-			email: null,
-			avatar: null,
-			githubURL: null,
-			permissionLevel: 'unknown',
-			currentUserSession: undefined,
-		};
-	}
-	return {
+	let user: authHelperResponse = {
 		id: '',
+		permissionLevel: 'unknown',
 		username: null,
 		name: null,
 		email: null,
 		avatar: null,
 		githubURL: null,
-		permissionLevel: 'unknown',
 		currentUserSession: undefined,
 	};
+
+	if (locals.isLoggedIn && locals.dbUser) {
+		let permissionLevel: 'admin' | 'editor' | 'visitor' | 'unknown' = 'unknown';
+		const permissions = await db
+			.select()
+			.from(Permissions)
+			.where(sql`lower(${Permissions.username}) = ${locals.dbUser.username.toLowerCase()}`)
+			.get();
+
+		if (
+			permissions &&
+			permissions.username.toLowerCase() === locals.dbUser.username.toLowerCase()
+		) {
+			switch (permissions.rank) {
+				case 'admin':
+					permissionLevel = 'admin';
+					break;
+				case 'editor':
+					permissionLevel = 'editor';
+					break;
+				case 'visitor':
+					permissionLevel = 'visitor';
+					break;
+				default:
+					permissionLevel = 'unknown';
+					break;
```

Comment:
Is rank ever anything other than these? Not really sure how the switch statement is helping

Reply:
1, Switch statement looks better than the previous if statements.  

2, No.  there is only 4 "ranks" currently.  and the permissions table is not actually directly "User-connected"  so this is also the first point within StudioCMS where "Ranks" are introduced and then used from.

Reply:
so you could just e.g. assign `permissionLevel = permissions.rank ?? 'unknown'`?

Reply:
i can't because i am using this to verify what is also in the DB falls within the dashboard's rules, since the end-user could modify the data directly in the database (its stored as a string value and we are not able to set enums for value control within `@astrojs/db` yet.  so its being used to both verify the current permissions as well as verify the user having a VALID permission config